underdocs-renderer/src/main/kotlin/underdocs/renderer/output/html/render/page/PageRenderer.kt

```diff
@@ -67,10 +67,16 @@ class PageRenderer(
                     .with(
                         UnescapedText(
                             """
-                                const isNightModePreferred = () => window.matchMedia?.('(prefers-color-scheme: dark)').matches;
-
+                                const getPreferredTheme = () => {
+                                  if (localStorage.getItem('underdocs_theme') === null) {
+                                    return window.matchMedia?.('(prefers-color-scheme: dark)').matches ? 'night' : 'day';
+                                  }
+                                
+                                  return localStorage.getItem('underdocs_theme');
```

Comment:
Maybe instead of this I could write something like this:

const preferredTheme = () => localStorage.getItem('underdocs_theme') === null ? window.matchMedia?.('(prefers-color-scheme: dark)').matches ? 'night' : 'day' : localStorage.getItem('underdocs_theme'); 

But I think the current one is more readable. Should I use the one-liner or it's better with the if-statement?

(Personally I prefer the if-statement)

Reply:
The `if` statement is much better, good choice!

Another thing: you can replace the `windows.matchMedia?...` thingy in line 72 with an `isNightModePreferred` call.

Reply:
Also, you can avoid the multiple `getItem` calls:

~~~~JavaScript
const savedTheme = localStorage.getItem()

if (savedTheme === null) {
}

return savedTheme
~~~~

Reply:
You are right, thanks for pointing out the repeated function calls.